pyrene/src/utils/DateUtils.ts

```diff
@@ -0,0 +1,264 @@
+ * @param {DateType} date
+ */
+export const isValidDate = (date?: DateType): boolean => {
+  if (date) {
+    return isExists(date.year, date.month - 1, date.day);
+  }
+  return false;
+};
+
+/**
+ * Checks if a timetype is valid
+ * @param {TimeType} time
+ */
+export const isValidTime = (time?: TimeType): boolean => {
+  if (time && time.hours >= 0 && time.hours <= 23 && time.minutes >= 0 && time.minutes <= 59) {
+    return true;
+  }
+  return false;
+};
+
+/** Checks if a timezone string is valid
+* @param {string} timezone
+*/
+export const isValidTimeZone = (timezone: string): boolean => {
+  try {
+    utcToZonedTime(getCurrentJsDateObject(), timezone);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const zeroFill = (num: string, length: number): string => (num.toString().padStart(length, '0'));
+
+export const convertDateTypeToString = (date: DateType): string => {
+  const day = zeroFill(date.day.toString(), 2);
+  const month = zeroFill(date.month.toString(), 2);
+  const year = zeroFill(date.year.toString(), 4);
+
+  return `${year}-${month}-${day}`;
+};
+
+export const convertTimeTypeToString = (time: TimeType): string => {
+  const hours = zeroFill(time.hours.toString(), 2);
+  const minutes = zeroFill(time.minutes.toString(), 2);
+
+  return `${hours}:${minutes}`;
+};
+
+export const dateTypeToStandardEUDateFormat = (date: DateType): string => {
+  const day = zeroFill(date.day.toString(), 2);
+  const month = zeroFill(date.month.toString(), 2);
+  const year = zeroFill(date.year.toString(), 4);
+
+  return `${day}.${month}.${year}`;
+};
+
+export const standardEUDateFormat = (date: Date): string => format(date, 'dd.MM.yyyy');
+
+export const standardEUTimeFormat = (date: Date): string => format(date, 'HH:mm');
+
+export const getDateTypeFromddmmyyyyWithSep = (str: string): DateType | undefined => {
+  if (str.length === 10 && allowedSeparatorCheck(str.charAt(2)) && allowedSeparatorCheck(str.charAt(5))) {
+    const date = { day: +str.substr(0, 2), month: +str.substr(3, 2), year: +str.substr(6) };
+    if (!Number.isNaN(date.day) && !Number.isNaN(date.month) && !Number.isNaN(date.year)) {
+      return date;
+    }
+  }
+  return undefined;
+};
+
+export const getTimeTypeFromhhmmWithSep = (str: string): TimeType | undefined => {
+  if (str.length === 5 && allowedSeparatorCheck(str.charAt(2))) {
+    const time = { hours: +str.substr(0, 2), minutes: +str.substr(3) };
+    if (!Number.isNaN(time.hours) && !Number.isNaN(time.minutes)) {
+      return time;
+    }
+  }
+  return undefined;
+};
+
+export const inRange = (timestampToCheck: number, minimumValue: number, maximumValue: number): number => {
+  if (timestampToCheck < minimumValue) {
+    return -1;
+  }
+  if (timestampToCheck > maximumValue) {
+    return 1;
+  }
+  return 0;
+};
+
+export const getErrors = (dateInvalid?: boolean, timeInvalid?: boolean, dateString?: string, minimumValue?: number, maximumValue?: number, timeZone?: string): string => {
```

Comment:
`getErrors` is very difficult to read. If the function take more than 3 params, it s [better](https://eslint.org/docs/rules/max-params) to pass an object.

Reply:
✅